module/VuFind/src/VuFind/Search/Factory/SolrDefaultBackendFactory.php

```diff
@@ -82,11 +91,23 @@ protected function getSolrCore()
     protected function createBackend(Connector $connector)
     {
         $backend = parent::createBackend($connector);
+        $backend->setRecordCollectionFactory($this->createRecordCollectionFactory());
+        return $backend;
+    }
+
+    /**
+     * Create the record collection factory.
+     *
+     * @return RecordCollectionFactoryInterface
+     */
+    protected function createRecordCollectionFactory()
+        : RecordCollectionFactoryInterface
+    {
         $manager = $this->serviceLocator
             ->get(\VuFind\RecordDriver\PluginManager::class);
-        $factory
-            = new RecordCollectionFactory([$manager, $this->createRecordMethod]);
-        $backend->setRecordCollectionFactory($factory);
-        return $backend;
+        return new RecordCollectionFactory(
```

Comment:
Should we also make RecordCollectionFactory a class property, so you can swap out the factory class without overriding the method?

Reply:
I didn't do it yet because createRecordCollectionFactory is pretty simple and the likelihood of needing to override the factory is small, but I'll be happy to make it that way.